src/mozanalysis/config.py

```diff
@@ -2,8 +2,8 @@
 # License, v. 2.0. If a copy of the MPL was not distributed with this
 # file, You can obtain one at http://mozilla.org/MPL/2.0/.
 
-from typing import List, Optional
 from dataclasses import dataclass
+from typing import List, Optional
```

Comment:
I thought it was odd to see `List` here because the `"UP"` lint rules should be suggesting "Use `list` instead of `List` for type annotation", and in fact I see that when I run `ruff` locally on main. I went to look at why [CI is passing](https://app.circleci.com/pipelines/github/mozilla/mozanalysis/845/workflows/807f45a9-919d-44f2-b4d6-366b119a9321/jobs/2572) with this still in here and it kind of looks like ruff isn't running? Maybe I'm just missing something..

Even more strange, I see that other files (like experiment.py) have removed `List`, `Dict`, etc. in favor of `list`, `dict`, etc. as expected.

None of this is a big problem, but I'm certainly confused and a little concerned about what's going on.

Reply:
Weirdly when I run locally ruff does not flag this and it runs fine, so maybe there is some sort of weird environment thing happening?

Reply:
Just to clarify the behavior I'm seeing... The `UP` rule error I quoted above appears for some files but not others. I don't remember if it was different last week when I looked (I suspect not), but right now I don't see the lint error for `config.py` where I commented, but if I go to `main` where the formatting changes haven't been made yet and run `ruff` with `UP` rules, it flags other files (the ones that you've already fixed). I don't see any `# noqa` or other indications that the rule should be ignored in `config.py`, so I don't know why this is happening.

That said, it's probably not worth blocking on this odd behavior. I'll finish reviewing the rest of the PR now that I'm back.

Reply:
Lol and now the CI is flagging the format thing, fixed

Reply:
Oh nice work! Still seems strange to me that it fixed some but not others, but glad you got it figured out.

Reply:
Yeah it is super weird, had to dig through issues on Ruff's github to finally figure it out 🙃